pipeline/extractor/src/main/scala/org/sunbird/obsrv/extractor/functions/ExtractionFunction.scala

```diff
@@ -37,26 +41,35 @@ class ExtractionFunction(config: ExtractorConfig, @transient var dedupEngine: De
                               context: ProcessFunction[mutable.Map[String, AnyRef], mutable.Map[String, AnyRef]]#Context,
                               metrics: Metrics): Unit = {
     metrics.incCounter(config.defaultDatasetID, config.totalEventCount)
+    val obsrvMeta = batchEvent.clone()(Constants.OBSRV_META).asInstanceOf[mutable.Map[String, AnyRef]]
```

Comment:
Can we avoid this clone object?

Reply:
Clone of object could be expensive operation.

Reply:
Or clone the object only when it's required.

Reply:
Removed cloning the object